File: benches/benches/primes.js
function primes(max_number) {
  const prime_mask = Array(max_number).fill(true);

  prime_mask[0] = false;
  prime_mask[1] = false;

  let total_primes_found = 0;

  for (let p = 2; p <= max_number; p += 1) {
    if (!prime_mask[p]) continue;

    total_primes_found += 1;

    let i = 2 * p;
    while (i < max_number + 1) {
      prime_mask[i] = false;
      i += p;
    }
  }

  return total_primes_found;
}

module.exports = { primes };

File: benches/benches/primes.lua
local _M = {}

function _M.primes(max_number)
  local prime_mask = {}
  for i = 1,max_number do
    prime_mask[i] = true
  end

  prime_mask[0] = false
  prime_mask[1] = false

  local total_primes_found = 0

  for p = 2, max_number do
    if not prime_mask[p] then goto continue end

    total_primes_found = total_primes_found + 1

    local i = 2 * p
    while i < max_number + 1 do
      prime_mask[i] = false
      i = i + p
    end

    ::continue::
  end

  return total_primes_found
end

return _M
